pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -407,23 +407,20 @@ func referencesDiffs(old []metav1.OwnerReference, new []metav1.OwnerReference) (
 	for _, value := range new {
 		newUIDToRef[string(value.UID)] = value
 	}
-	newUIDSet := sets.StringKeySet(newUIDToRef)
-
-	addedUID := newUIDSet.Difference(oldUIDSet)
-	removedUID := oldUIDSet.Difference(newUIDSet)
-	intersection := oldUIDSet.Intersection(newUIDSet)
-
-	for uid := range addedUID {
-		added = append(added, newUIDToRef[uid])
-	}
-	for uid := range removedUID {
-		removed = append(removed, oldUIDToRef[uid])
-	}
-	for uid := range intersection {
-		if !reflect.DeepEqual(oldUIDToRef[uid], newUIDToRef[uid]) {
-			changed = append(changed, ownerRefPair{oldRef: oldUIDToRef[uid], newRef: newUIDToRef[uid]})
+	for newUID := range newUIDToRef {
```

Comment:
Since each element of newUIDToRef is accessed exactly once in this loop, why not eliminate the map entirely and simply loop over new here? (But see the overall comment about benchmarks first.)